src/lazy/binary/encoded_value.rs

```diff
@@ -130,6 +128,15 @@ impl<HeaderType: EncodedHeader> EncodedValue<HeaderType> {
         start..end
     }
 
+    /// Returns the number of bytes used to encode this value's opcode. If this value was serialized
+    /// using a tagless encoding, returns `0`.
+    pub fn opcode_length(&self) -> usize {
+        match self.encoding {
+            BinaryValueEncoding::Tagged => 1,
+            _ => 0,
+        }
+    }
+
```

Comment:
Rather that setting/storing a number of opcode bytes on every value, we can derive whether an opcode byte was present. Fixes #805.